sentry/src/main/java/io/sentry/SentryReplayEvent.java

```diff
@@ -0,0 +1,324 @@
+  }
+
+  public void setSegmentId(final int segmentId) {
+    this.segmentId = segmentId;
+  }
+
+  @NotNull
+  public Date getTimestamp() {
+    return timestamp;
+  }
+
+  public void setTimestamp(final @NotNull Date timestamp) {
+    this.timestamp = timestamp;
+  }
+
+  @Nullable
+  public Date getReplayStartTimestamp() {
+    return replayStartTimestamp;
+  }
+
+  public void setReplayStartTimestamp(final @Nullable Date replayStartTimestamp) {
+    this.replayStartTimestamp = replayStartTimestamp;
+  }
+
+  @Nullable
+  public List<String> getUrls() {
+    return urls;
+  }
+
+  public void setUrls(final @Nullable List<String> urls) {
+    this.urls = urls;
+  }
+
+  @Nullable
+  public List<String> getErrorIds() {
+    return errorIds;
+  }
+
+  public void setErrorIds(final @Nullable List<String> errorIds) {
+    this.errorIds = errorIds;
+  }
+
+  @Nullable
+  public List<String> getTraceIds() {
+    return traceIds;
+  }
+
+  public void setTraceIds(final @Nullable List<String> traceIds) {
+    this.traceIds = traceIds;
+  }
+
+  @NotNull
+  public ReplayType getReplayType() {
+    return replayType;
+  }
+
+  public void setReplayType(final @NotNull ReplayType replayType) {
+    this.replayType = replayType;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    SentryReplayEvent that = (SentryReplayEvent) o;
+    return segmentId == that.segmentId
+        && Objects.equals(type, that.type)
+        && replayType == that.replayType
+        && Objects.equals(replayId, that.replayId)
+        && Objects.equals(urls, that.urls)
+        && Objects.equals(errorIds, that.errorIds)
+        && Objects.equals(traceIds, that.traceIds);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(type, replayType, replayId, segmentId, urls, errorIds, traceIds);
+  }
+
+  // region json
+  public static final class JsonKeys {
+    public static final String TYPE = "type";
+    public static final String REPLAY_TYPE = "replay_type";
+    public static final String REPLAY_ID = "replay_id";
+    public static final String SEGMENT_ID = "segment_id";
+    public static final String TIMESTAMP = "timestamp";
+    public static final String REPLAY_START_TIMESTAMP = "replay_start_timestamp";
+    public static final String URLS = "urls";
+    public static final String ERROR_IDS = "error_ids";
+    public static final String TRACE_IDS = "trace_ids";
+  }
+
+  @Override
+  @SuppressWarnings("JdkObsolete")
+  public void serialize(final @NotNull ObjectWriter writer, final @NotNull ILogger logger)
+      throws IOException {
+    writer.beginObject();
+    writer.name(JsonKeys.TYPE).value(type);
+    writer.name(JsonKeys.REPLAY_TYPE).value(logger, replayType);
+    writer.name(JsonKeys.SEGMENT_ID).value(segmentId);
+    writer
+        .name(JsonKeys.TIMESTAMP)
+        .value(logger, BigDecimal.valueOf(DateUtils.dateToSeconds(timestamp)));
```

Comment:
Yeah the reason was that the milliseconds precision is lost (e.g. `12345.678` will transform to something like `12345.6E10`) when using plain double and converting it to BigDecimal helped. But I think it's not necessary actually, because we can just send ISO Date and relay should take care of converting it to whatever is needed for frontend, so I'll change that 👍